support/android/apk/build.gradle

```diff
@@ -17,6 +18,65 @@ allprojects {
         }
         google()
     }
+}
+
+// Utility methods
+String getTargetDir(boolean debug, String arch) {
+    def basePath = project.rootDir.getParentFile().getParentFile().getParentFile().absolutePath
+    return basePath + '/target/' + getSubTargetDir(debug, arch)
+}
+
+String getSubTargetDir(boolean debug, String arch) {
+    return getRustTarget(arch) + '/' + (debug ? 'debug' : 'release')
+}
+
+String getJniLibsPath(boolean debug, String arch) {
+    return getTargetDir(debug, arch) + '/apk/jniLibs'
+}
+
+static String getRustTarget(String arch) {
+    switch (arch.toLowerCase()) {
+        case 'arm' : return 'arm-linux-androideabi'
+        case 'armv7' : return 'armv7-linux-androideabi'
+        case 'arm64' : return 'aarch64-linux-android'
+        case 'x86' : return 'i686-linux-android'
+        default: throw new GradleException("Invalid target architecture " + arch)
+    }
+}
+
+static String getNDKAbi(String arch) {
+    switch (arch.toLowerCase()) {
+        case 'arm' : return 'armeabi'
+        case 'armv7' : return 'armeabi-v7a'
+        case 'arm64' : return 'arm64-v8a'
+        case 'x86' : return 'x86'
+        default: throw new GradleException("Invalid target architecture " + arch)
+    }
+}
+
+String getNdkDir() {
+    // Read environment variable used in rust build system
+    String ndkDir = System.getenv('ANDROID_NDK')
+    if (ndkDir == null) {
+        ndkDir = System.getenv('ANDROID_NDK_HOME')
```

Comment:
We should also check ANDROID_NDK_ROOT. Some android devs use that: https://bugs.swift.org/browse/SR-1365